src/script/main/app.js

```diff
@@ -734,7 +734,16 @@ class App {
       // Clear Local Storage (but don't delete the cookie label if you were logged in with a permanent client)
       const keysToKeep = [StorageKey.AUTH.SHOW_LOGIN];
 
-      const keepPermanentDatabase = this.repository.client.isCurrentClientPermanent() && !clearData;
+      let keepPermanentDatabase = !clearData;
+
+      try {
+        keepPermanentDatabase = this.repository.client.isCurrentClientPermanent() && !clearData;
+      } catch (error) {
+        if (error.type === z.error.ClientError.TYPE.CLIENT_NOT_SET) {
+          keepPermanentDatabase = false;
+        }
```

Comment:
If error is not of type `CLIENT_NOT_SET`, should we throw it then?

Reply:
No, then we should go with whatever is set to `keepPermanentDatabase` in this stage.